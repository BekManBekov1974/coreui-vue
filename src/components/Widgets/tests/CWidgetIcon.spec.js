import { shallowMount } from '@vue/test-utils'
import Component from '../CWidgetIcon'
import CLink from '../../Link/CLink'
import { CIcon } from '@coreui/icons/vue'

import Vue from 'vue'

Vue.component('CLink', CLink)
Vue.component('CIcon', CIcon)

const ComponentName = 'CWidgetIcon'
const defaultWrapper = shallowMount(Component)
const customWrapper = shallowMount(Component, {
  propsData: {
    header: 'header',
    text: 'text',
    link: 'link.com',
    addIconClasses: 'someClasses'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  // it('correctly render component with default props', () => {
  //   expect(defaultWrapper.findAll('div').at(2).text()).toMatch('header:string')
  //   expect(defaultWrapper.findAll('div').at(3).text()).toMatch('text:string')
  //   expect(defaultWrapper.find('i').attributes().class).toMatch('fa fa-cogs bg-primary')
  //   expect(defaultWrapper.find('clink-stub').exists()).toBe(false)
  // })
  it('matches default props snapshot', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  // it('correctly render component with custom props', () => {
  //   expect(customWrapper.findAll('div').at(2).text()).toMatch('header')
  //   expect(customWrapper.findAll('div').at(3).text()).toMatch('text')
  //   expect(customWrapper.find('i').attributes().class).toMatch('someClasses')
  //   expect(customWrapper.find('clink-stub').exists()).toBe(true)
  //   expect(customWrapper.find('clink-stub').attributes().href).toMatch('link.com')
  // })
  it('matches custom props snapshot', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
