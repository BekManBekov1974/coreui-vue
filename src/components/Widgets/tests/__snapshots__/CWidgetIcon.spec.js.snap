// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidgetIcon matches custom props snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body clearfix d-flex align-items-center"
  >
    <i
      class="float-left mr-3 font-2xl p-3"
    />
     
    <div>
      <div
        class="text-primary text-value-sm"
      >
        
        header
      
      </div>
       
      <div
        class="text-muted text-uppercase font-weight-bold small"
      >
        
        text
      
      </div>
    </div>
  </div>
   
  <div
    class="card-footer px-3 py-2"
  >
    <a
      class="font-weight-bold font-xs btn-block text-muted"
      href="link.com"
      target="_self"
    >
      
        View More
        
      <i
        class="fa fa-angle-right float-right font-lg"
      />
    </a>
  </div>
</div>
`;

exports[`CWidgetIcon matches default props snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body clearfix d-flex align-items-center"
  >
    <i
      class="float-left mr-3 font-2xl p-3"
    />
     
    <div>
      <div
        class="text-primary text-value-sm"
      >
        
        header:string
      
      </div>
       
      <div
        class="text-muted text-uppercase font-weight-bold small"
      >
        
        text:string
      
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
